src/adapters/github.js

```diff
@@ -342,16 +347,22 @@ class GitHub extends PjaxAdapter {
           const hugeRepos = this.store.get(STORE.HUGE_REPOS);
           const repo = `${opts.repo.username}/${opts.repo.reponame}`;
           const repos = Object.keys(hugeRepos);
-          if (!hugeRepos[repo]) {
-            // If there are too many repos memoized, delete the oldest one
-            if (repos.length >= GH_MAX_HUGE_REPOS_SIZE) {
-              const oldestRepo = repos.reduce((min, p) => (hugeRepos[p] < hugeRepos[min] ? p : min));
-              delete hugeRepos[oldestRepo];
-            }
-            hugeRepos[repo] = new Date().getTime();
-            this.store.set(STORE.HUGE_REPOS, hugeRepos);
+
+          const handleHugeRepo = (isFullStorage) => this._handleError(cfg, {status: isFullStorage ? 600 : 206}, cb);
+
+          // The hugeRepos isn't a default {} due to the full Local Storage
+          if (!hugeRepos) return handleHugeRepo(true);
```

Comment:
Why should store.get() return wrong data when storage is full? Are you sure about this?

Reply:
When the new user installs or update the Octotree v3 but his local storage is already full, the default setting is unable to set storage, which leads to the variable is undefined

Reply:
In this case, should we just short circuit the code to return early, skipping the check huge repo logic? That would make things simple without a big and ugly change like this (now if it's less ugly, I'm okay, but this is really ugly).